hoomd/pytest/test_simulation.py

```diff
@@ -239,6 +239,49 @@ def test_state_from_gsd(device, simulation_factory, lattice_snapshot_factory,
         assert_equivalent_snapshots(snap, sim.state.get_snapshot())
 
 
+@skip_gsd
+def test_state_from_gsd_box_dims(device, simulation_factory,
+                                 lattice_snapshot_factory, tmp_path):
+
+    def modify_gsd_snap(gsd_snap):
+        """Add nonzero z values to gsd box for testing."""
+        new_box = list(gsd_snap.configuration.box)
+        new_box[2] = 1e2 * (np.random.random() - 0.5)
+        new_box[4] = 1e2 * (np.random.random() - 0.5)
+        new_box[5] = 1e2 * (np.random.random() - 0.5)
+        gsd_snap.configuration.box = new_box
+        return gsd_snap
+
+    d = tmp_path / "sub"
+    d.mkdir()
+    filename = d / "temporary_test_file.gsd"
+    if device.communicator.rank == 0:
+        f = gsd.hoomd.open(name=filename, mode='wb+')
+
+    sim = simulation_factory(
+        lattice_snapshot_factory(n=10, particle_types=["A", "B"], dimensions=2))
+    snap = sim.state.get_snapshot()
+
+    if device.communicator.rank == 0:
+        f.append(modify_gsd_snap(make_gsd_snapshot(snap)))
+
+    checks = range(3)
+    for step in checks:
+        if device.communicator.rank == 0:
+            f.append(modify_gsd_snap(make_gsd_snapshot(snap)))
+
+    if device.communicator.rank == 0:
+        f.close()
```

Comment:
This can be written so there is only 1 `if device.communicator.rank == 0` check.